lib/dry/types/constructor.rb

```diff
@@ -71,6 +71,13 @@ def constrained_type
         Constrained::Coercible
       end
 
+      # @api public
+      #
+      # @see Definition#to_ast
+      def to_ast
```

Comment:
I have some questions regarding the `Constructor` `#ast`, since it have `fn` that tells us information regarding the `coercion` method, I do not know if is important to included in the `ast`. Something like `[[:type, type.to_ast], [:fn, fn.to_s]]` id not really clear if we would be able to extract info from the `fn`